lib/primitives/array-element.js

```diff
@@ -93,7 +96,12 @@ module.exports = function(BaseElement, registry) {
     },
 
     set: function(index, value) {
-      this.content[index] = registry.toElement(value);
+      var item = registry.toElement(value);
+      if (!_.contains(item.parents, this)) {
+        item.parents.push(this);
+      }
+      this.content[index] = item;
+      this.emitSwap();
```

Comment:
swap? Sort of confusing semantics. This is replacing the element at some position with another, is that correct?